hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/validate/types/HazelcastTypeUtils.java

```diff
@@ -103,7 +103,8 @@ public static SqlTypeName toCalciteType(QueryDataTypeFamily typeFamily) {
     }
 
     public static QueryDataType toHazelcastType(SqlTypeName sqlTypeName) {
-        switch (sqlTypeName.getFamily()) {
+        // some types have null family (e.g. SqlTypeName.ROW), we fall back to default branch for those
+        switch (sqlTypeName.getFamily() != null ? sqlTypeName.getFamily() : SqlTypeFamily.ANY) {
```

Comment:
That's a separate issue. The other thing is, personally I find #18437 a more readable.

Reply:
Removed it